src/main/java/pl/edu/amu/szi/forklift/ForkliftController.java

```diff
@@ -96,6 +90,54 @@ else if(currentKeys.contains("I"))
         }
     }
 
+    public void putOnShelf(Map map)
+    {
+        if (map.packageFound(forklift.getXPos(), forklift.getYPos())) {
+            if(!map.isPassable(forklift.getXPos(),forklift.getYPos()-1)){ // check if shelf is above
```

Comment:
isPassable nie zadziała czasem też dla brzegów mapy? Może lepiej byłoby użyć [instanceof](https://stackoverflow.com/questions/7526817/use-of-instance-of-in-java)